codec/abi_decoder.go

```diff
@@ -598,16 +598,24 @@ func (d *ABIDecoder) decodeAction(action *pbantelope.Action, globalSequence uint
 		return nil
 	}
 
-	action.JsonData = strings.ToValidUTF8(string(jsonData), "�")
-	zlog.Debug("successfully decoded action data", zap.String("action", action.SimpleName()), zap.String("json_data", string(jsonData)))
+	if flags.HasFeatureEnabled(flags.FeatureUltra) {
```

Comment:
Do we need this flag though since we already fork firehose-antelope, we can just implement inject action
And for `inject` action, i don't think we need it anymore cause this action is from `migrator` contract which we don't use anymore
You can check the migrator here https://github.com/ultraio/dfuse-eosio/blob/b1d3d18c416f1bb9e26d41a1d3e2c6cc74318e88/booter/migrator/code/migrator.cpp#L20

Reply:
Restored the whole file.
The other updated parts are also from the old code, I reviewed them again and found that the later code is better for all updates.